crates/bevy_gizmos/src/rounded_box.rs

```diff
@@ -0,0 +1,379 @@
+                Vec3::X,
+                Vec2::new(self.size.z, self.size.y),
+                Quat::from_rotation_y(FRAC_PI_2),
+            ),
+            (
+                Vec3::Y,
+                Vec2::new(self.size.x, self.size.z),
+                Quat::from_rotation_x(FRAC_PI_2),
+            ),
+            (Vec3::Z, Vec2::new(self.size.x, self.size.y), Quat::IDENTITY),
+        ];
+
+        for (position, size, rotation) in rects {
+            let world_rotation = config.rotation * rotation;
+            let local_position = config.rotation * (position * inner_half_size);
+            self.gizmos
+                .rounded_rect(
+                    config.position + local_position,
+                    world_rotation,
+                    size,
+                    config.color,
+                )
+                .arc_segments(config.arc_segments)
+                .corner_radius(edge_radius);
+
+            self.gizmos
+                .rounded_rect(
+                    config.position - local_position,
+                    world_rotation,
+                    size,
+                    config.color,
+                )
+                .arc_segments(config.arc_segments)
+                .corner_radius(edge_radius);
+        }
+    }
+}
+
+impl<'w, 's, T: GizmoConfigGroup> Gizmos<'w, 's, T> {
+    /// Draw a wireframe rectangle with rounded corners in 3D.
+    ///
+    /// This should be called for each frame the rectangle needs to be rendered.
+    ///
+    /// # Arguments
+    ///
+    /// - `position`: The center point of the rectangle.
+    /// - `rotation`: defines orientation of the rectangle, by default we assume the rectangle is contained in a plane parallel to the XY plane.
+    /// - `size`: defines the size of the rectangle. This refers to the 'outer size', similar to a bounding box.
+    /// - `color`: color of the rectangle
+    ///
+    /// # Builder methods
+    ///
+    /// - The corner radius can be adjusted with the `.corner_radius(...)` method.
+    /// - The number of segments of the arcs at each corner (i.e. the level of detail) can be adjusted with the
+    /// `.arc_segments(...)` method.
+    ///
+    /// # Example
+    /// ```
+    /// # use bevy_gizmos::prelude::*;
+    /// # use bevy_render::prelude::*;
+    /// # use bevy_math::prelude::*;
+    /// # use bevy_color::palettes::css::GREEN;
+    /// fn system(mut gizmos: Gizmos) {
+    ///     gizmos.rounded_rect(
+    ///         Vec3::ZERO,
+    ///         Quat::IDENTITY,
+    ///         Vec2::ONE,
+    ///         GREEN
+    ///         )
+    ///         .corner_radius(0.25)
+    ///         .arc_segments(10);
+    /// }
+    /// # bevy_ecs::system::assert_is_system(system);
+    /// ```
+    pub fn rounded_rect(
+        &mut self,
+        position: Vec3,
+        rotation: Quat,
+        size: Vec2,
+        color: impl Into<Color>,
+    ) -> RoundedRectBuilder<'_, 'w, 's, T> {
+        let corner_radius = size.min_element() / 10.0;
```

Comment:
Maybe we should pull the default of `10.0` into a constant? Both to make it clear where it's coming from, and to split up the need to tweak the implementation if in the future the default needs to change.

I'm not 100% sure if we could also reference it from the docs, but that could help us keep a single definition for the default.

EDIT: Just saw we're doing this for `DEFAULT_ARC_SEGMENTS` already 👍 Could be handy to give corner radius the same treatment.

Reply:
Good suggestion :)